### ISSUE TITLE
Leap software development

### ISSUE THREAD

**Issue by chillsauce** [EOS-Nation/mandel#41]
As an EOS node operator, I need to know what version of EOSIO I should be testing on testnets so that I can be prepared for the eventual upgrade on mainnet. 

- or 3.1 for everyone
- Release candidate target mid-late June
- Final release about a month later
- aim to have BPs upgraded by end of August
- consensus upgrade can happen later (September 21)

Useful consensus features to consider activating by fall:
- configurable WASM limits (careful - incorrect config could brick the chain. test all configurations on testnets)
- get code hash
- return value from action
   - configure blockchain params 

- [ ] Need to align on what consensus features we will activate

**Comment by chillsauce** [comment]
Transaction lifecycle features may be ready to start testing in late May

**Comment by chillsauce** [comment]
ENF will be sharing the acceptance criteria broadly for the transaction lifecycle features in the next week

**Comment by chillsauce** [comment]
Question from @matthewdarwin in Telegram:

I just want to make sure I understand what is in mandel 3.1.  From what I heard, it has the following things:

- based on mandel 3.0rc1
- merge in b1 2.0 latest stuff https://github.com/eosnetworkfoundation/mandel/pull/43
- port dfuse support from 2.1 https://github.com/eosnetworkfoundation/mandel/pull/38
- port blocks.log rotation feature (strides) from 2.1
- transaction lifecycle features (4 of them, as per API+ blue paper)

Anything else planned so far?

**Comment by chillsauce** [comment]
The following EOSIO 2.1 items added into Mandel 3.1 (alongside what is already built in
Mandel 3.0):
● Provide options for nodeos to automatically split the blocks log and state history logs
(every X blocks).
● An eosio_blocklog utility that can read and split the logs.
● Merge the work of the dfuse team (deep-mind logger) that was included in the 2.1
release.
● Remove deprecated features and modules listed at EOSIO Deprecations · Issue
#7597 · EOSIO/eos (github.com).
https://github.com/EOSIO/eos/issues/7597

**Comment by chillsauce** [comment]
We're waiting for confirmation from Fractally team that they will commit to the 3.1 work

**Comment by chillsauce** [comment]
3.0.5 has been merged and is ready to be tested on Jungle
https://github.com/eosnetworkfoundation/mandel/pull/38

**Comment by chillsauce** [comment]
3.0.5.RC1 is now tagged with release notes

**Comment by chillsauce** [comment]
Request for another feature to backport: https://github.com/eosnetworkfoundation/mandel/issues/85

**Comment by chillsauce** [comment]
From @matthewdarwin :

Hello all,

As part of mandel 3.1 plans, it has been previously discussed that the "blocks.log splitting feature" is a necessary requirement for the release to be acceptable to the community. Recently, a question arose, as to what is the actual requirements necessary for 3.1 related to this feature.

It is my (and a few other people's view) that the desirable list of features for mandel 3.1 is as follows:

1) keep all the blogs.log, in a single file, as per nodeos 2.0 (no change). Starts from beginning or based on a snapshot.

2) keep some recent blocks, in one file (or optionally more files at the discretion of the developers),  and remove old blocks when the configured number of blocks passes
  - this is less features than is in 2.1 where it can split files, move them to archive directory, and other things

3) don't store any blocks at all (this is a new feature, not in 2.1)

If you have opinions one way or another, please add thoughts to the github issues:
https://github.com/eosnetworkfoundation/mandel/issues/100
https://github.com/eosnetworkfoundation/mandel/issues/92

Above also applies to "ship" (state history)

**Comment by chillsauce** [comment]
- Transaction lifecycle features are completed
- Long list of bug fixes have been completed and are being documented
- Block log splitting in progress, will be delivered close to June 1
- 80% complete analysis on 2.1 and 2.2 features and all PRs and prioritized and logged issues for Mandel (2500 PRs)
- Mandel 3.1 contains all of 2.0 and the key features of 2.1 plus a bunch of bug fixes

**Comment by interwebcoding** [comment]
> The following EOSIO 2.1 items added into Mandel 3.1 (alongside what is already built in Mandel 3.0):
> 
> * Provide options for nodeos to automatically split the blocks log and state history logs
>   (every X blocks).
> * An eosio_blocklog utility that can read and split the logs.
> * ~Merge the work of the dfuse team (deep-mind logger) that was included in the 2.1
>   release.~
> * Remove deprecated features and modules listed at [EOSIO Deprecations EOSIO/eos#7597](https://github.com/EOSIO/eos/issues/7597)

Hi, just a question about 2.1 Key Value tables.   We have an enterprise customer using this feature.  Last I checked, there was going to be a path in Mandel where this would be supported.  Please could you confirm if this is still the case?
https://eos.io/news/eosio-2-1/#key-value
Thanks  (Rewired.one)
@chillsauce

**Comment by chillsauce** [comment]
- planning for 3.1.0-rc2
  - will contain changes for EVM and SHIP, plus other stuff
    (SHIP is Fred's issue)
  - no release date set... probably in a week or so

- would release mandel-cdt rc1 at around the same time

- mandel-contracts rc1 would come later

**Comment by chillsauce** [comment]
Aiming to release RC2 with EVM and SHIP fixes next Tuesday, we will also require an RC3 to follow TBD date.

**Comment by chillsauce** [comment]
- RC2 will be released today, but still requires testing before we can activate the consensus features. 
- BPs asked to upgrade and test RC2 this week - need to validate we have a working history solution.
- Go/no-go for consensus feature activation next week
- Improved support and documentation in RC2 for Ubuntu 18.04 
- ENF is looking for feedback if FillPG is something that should be updated and maintained

**Comment by chillsauce** [comment]
- Green light from ENF to activate and test crypto primitives 
- CDT and contracts are released with release notes on Thursday
- RC3 will be required and targeting release next week ([includes fix for deep mind/dfuse](https://github.com/eosnetworkfoundation/mandel/pull/654))

**Comment by chillsauce** [comment]
- RC3 was a no-go for this week, planning for next week (hopefully this will become final)
- Removing library dependency on GMP library was moved out of scope from 3.1 to 3.2
- Need more testing on contracts and CDT

**Comment by chillsauce** [comment]
Mandel 3.1 RC3 release is imminent - will likely become final assuming no issues found on Jungle after a week of testing

**Comment by chillsauce** [comment]
ENF are reviewing a few minor issues that have been reported for Mandel 3.1 RC3 and are meeting today to discuss the plan to address them and timelines. It is likely they are minor enough to go straight into the final release.

**Comment by chillsauce** [comment]
- Mandel 3.1 RC4 and CDT 3.0 RC2 will be released today
- Small system contracts update coming tomorrow
- We should have a confident timeline on Mandel 3.1 final release next week

**Comment by chillsauce** [comment]
- Leap 3.1 Final release is out: https://github.com/AntelopeIO/leap/releases/tag/v3.1.0
- DUNE RC3 and CDT RC3 is expected within the next week or so

**Comment by chillsauce** [comment]
- [CDT RC3](https://github.com/AntelopeIO/cdt/releases/tag/v3.0.0-rc3) was released yesterday
- The current release of DUNE is not compatible with CDT RC3 and will require a minor update.

**Comment by stephenpdiesel** [comment]
DUNE RC3 has been released to support CDT RC3 and Leap 3.1.0 Stable https://github.com/AntelopeIO/DUNE/releases/tag/v1.0.0-rc3

**Comment by chillsauce** [comment]
- EOS system contracts, DUNE and CDT will likely go to Final by the end of the week. No changes expected, just a version bump.

**Comment by chillsauce** [comment]
- Patch release of 3.1.2 expected later today or by end of week
- 3.2.0 RC1 also expected by end of week

**Comment by chillsauce** [comment]
- [Leap 3.2 RC1 is out](https://github.com/AntelopeIO/leap/releases/tag/v3.2.0-rc1)
   - This should include all the best of 2.1 plus some improvements in particular for performance issues experienced on WAX
- CDT release 3.1 by end of this week or early next week
- Possibility we'll get EOS system contract release in the next 1-2 weeks

**Comment by stephenpdiesel** [comment]
We are actively addressing the msig issue mentioned in the last two calls within this issue. We do not have a definitive timeline for eos-system-contracts v3.2.0 but this will be part of it https://github.com/AntelopeIO/reference-contracts/issues/7. Please note this issue is for reference contracts but the change will be in both places.

**Comment by chillsauce** [comment]
- Expecting Leap 3.2 RC2 with some bug fixes by the end of this week or early next week
- CDT release 3.1 by end of this week or early next week
- EOS System contracts and reference contracts to be updated shortly after to make it more user friendly 
- ENF is still working towards a possible March/April timeline for next consensus upgrade to Leap 4.0

**Comment by chillsauce** [comment]
On the way:
- Leap 3.1.3 patch release
- Leap 3.2.0 RC2 
- CDT 3.1.0 RC1 for CDT
- System contract updates
- Release of DUNE in the next month

**Comment by chillsauce** [comment]
- Antelope software updates just released:
   - [Leap 3.1.3 patch release](https://github.com/AntelopeIO/leap/releases/tag/v3.1.3)
   - [Leap 3.2.0 RC2](https://github.com/AntelopeIO/leap/releases/tag/v3.2.0-rc2) 
- Antelope software updates on the way:
   - CDT 3.1.0 RC1
   - System contract updates
   - Release of DUNE in the next month
